tests/queries/test_query.py

```diff
@@ -165,6 +179,18 @@ def test_equality(self):
             Author.objects.filter(Q(item__name='foo')).query,
         )
 
+    def test_equality_with_negated_nullable(self):
+        # nullable field
+        self.assertNotEqual(
+            Item.objects.exclude(modified__year='2021').values('id').query,
+            Item.objects.annotate(year=ExtractYear('modified')).exclude(year='2021').values('id').query,
+        )
+        # not nullable field
+        self.assertNotEqual(
+            Item.objects.exclude(created__year='2021').values('id').query,
+            Item.objects.annotate(year=ExtractYear('created')).exclude(year='2021').values('id').query,
+        )
```

Comment:
This test passes without applying your patch -- we like tests that increase coverage to be in separate commits. (what if we have to revert the new behavior? we can keep the test increasing coverage.)

Reply:
@jacobtylerwalls Thank you for your kind review! 
I wanna do test equal... it was typo. i will fix it.

Reply:
I want to check this. I will fix this.
        self.assertEqual(
            str(Item.objects.exclude(modified__year='2021').values('id').query),
            str(Item.objects.annotate(year=ExtractYear('modified')).exclude(year='2021').values('id').query),
        )